translate-pixiv-tags.user.js

```diff
@@ -472,28 +593,121 @@ function get (url, params, cache = CACHE_LIFETIME, baseUrl = BOORU) {
         });
 }
 
+function queueNetworkRequest (type, item) {
+    const request = {
+        type,
+        item,
+        promise: $.Deferred(),
+    };
+    QUEUED_NETWORK_REQUESTS.push(request);
+    return request.promise;
+}
+
+const queueNetworkRequestMemoized = _.memoize(queueNetworkRequest, memoizeKey);
+
+function intervalNetworkHandler () {
+    Object.keys(NETWORK_REQUEST_DICT).forEach((type) => {
+        const requests = QUEUED_NETWORK_REQUESTS.filter((request) => (request.type === type));
+        if (requests.length > 0) {
+            const items = requests.map((request) => request.item);
+            const typeParam = NETWORK_REQUEST_DICT[type].params(items);
+            const params = Object.assign(typeParam, LIMIT_PARAM, CACHE_PARAM);
+            const url = `${BOORU}${NETWORK_REQUEST_DICT[type].url}.json`;
+            getLong(url, params, requests, type);
+        }
+    });
+    // Clear the queue once all network requests have been sent
+    QUEUED_NETWORK_REQUESTS.length = 0;
+}
+
+async function getLong (url, params, requests, type) {
+    const sleepHalfSecond = (resolve) => setTimeout(resolve, 500);
+    const domain = new URL(url).hostname;
+    if (!(checkNetworkErrors(domain, false))) {
+        requests.forEach((request) => request.promise.resolve([]));
+        return;
+    }
+
+    // Default to GET requests
+    let func = $.getJSON;
+    let finalParams = params;
+    if ($.param(params).length > MAXIMUM_URI_LENGTH) {
+        // Use POST requests only when needed
+        finalParams = Object.assign(finalParams, { _method: "get" });
+        func = $.post;
+    }
+
+    /* eslint-disable no-await-in-loop */
+    let resp = [];
+    for (let i = 0; i < MAX_NETWORK_RETRIES; i++) {
+        try {
+            resp = await func(url, finalParams);
+            break;
+        } catch (error) {
+            console.error(
+                "Failed try #",
+                i + 1,
+                "\nURL:",
+                url,
+                "\nParameters:",
+                finalParams,
+                "\nHTTP Error:",
+                error.status,
+            );
+            if (!checkNetworkErrors(domain, true)) {
+                requests.forEach((request) => request.promise.resolve([]));
+                return;
+            }
+            await new Promise(sleepHalfSecond);
+        }
+    }
+    /* eslint-enable no-await-in-loop */
+
+    let finalResp = resp;
+    if (NETWORK_REQUEST_DICT[type].filter) {
+        // Do any necessary filtering after the network request completes
+        finalResp = NETWORK_REQUEST_DICT[type].filter(resp);
+    }
+
+    // Get the data key which is used to find successful hits in the batch results
+    const dataKey = NETWORK_REQUEST_DICT[type].data_key;
+    requests.forEach((request) => {
```

Comment:
Here would nice to add some mechanism that checks and warns about responses which no one request used.

Reply:
I'm not sure what you're asking here. Are you talking about results that don't match any request? I'm not sure that is possible with the search parameters being used.

The other URL one, i.e. "url_matches", on artists was buggy because it did a lot of internal processing where it continuously slices up the URL and searches until it finds some results, whereas the ones that were added do just a one-to-one comparison.

So for example, if the following isn't true...

`lowercase(databaseItem) == lowercase(requestItem)`

... then it won't return that result. In fact it will probably be possible to straight up remove the fix for #18 after some transitory period.

However, I'll go ahead and add some checking in just so that fact can be validated.

Reply:
It's mainly to ensure that TRANSLATE_PROFILE_URL doesn't lack normalisations. You can make it debug only.